osu.Game/Screens/Select/BeatmapInfoWedgeV2.cs

```diff
@@ -0,0 +1,346 @@
+                        Colour = Colour4.Transparent,
+                        Depth = float.MaxValue,
+                        Anchor = Anchor.TopRight,
+                        Origin = Anchor.TopRight,
+                        RelativeSizeAxes = Axes.Y,
+
+                        // By limiting the width we avoid this box showing up as an outline around the drawables that are on top of it.
+                        Width = colour_bar_width + corner_radius,
+                        Child = new Box { RelativeSizeAxes = Axes.Both }
+                    },
+                    starCounter = new StarCounter
+                    {
+                        Colour = Colour4.Transparent,
+                        Anchor = Anchor.CentreRight,
+                        Origin = Anchor.Centre,
+                        Scale = new Vector2(0.35f),
+                        Shear = -wedged_container_shear,
+                        X = -colour_bar_width / 2,
+                        Direction = FillDirection.Vertical
+                    }
+                }
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            ruleset.BindValueChanged(_ => updateDisplay());
+
+            float starAngle = (float)(Math.Atan(shear_width / wedge_height) * (180 / Math.PI));
+
+            // Applying the rotation directly to the StarCounter distorts the stars, hence it is applied to the child container
+            starCounter.Children.First().Rotation = starAngle;
+
+            // Makes sure the stars center themselves properly in the colour bar
+            starCounter.Children.First().Anchor = Anchor.Centre;
+            starCounter.Children.First().Origin = Anchor.Centre;
+        }
+
+        private const double animation_duration = 800;
+
+        protected override void PopIn()
+        {
+            this.MoveToX(0, animation_duration, Easing.OutQuint);
+            this.RotateTo(0, animation_duration, Easing.OutQuint);
+            this.FadeIn(transition_duration);
+        }
+
+        protected override void PopOut()
+        {
+            this.MoveToX(-100, animation_duration, Easing.In);
+            this.RotateTo(10, animation_duration, Easing.In);
+            this.FadeOut(transition_duration * 2, Easing.In);
+        }
+
+        private WorkingBeatmap? beatmap;
+
+        public WorkingBeatmap? Beatmap
+        {
+            get => beatmap;
+            set
+            {
+                if (beatmap == value) return;
+
+                beatmap = value;
+
+                updateDisplay();
+            }
+        }
+
+        public override bool IsPresent => base.IsPresent || DisplayedContent == null; // Visibility is updated in the LoadComponentAsync callback
+
+        private Container? loadingInfo;
+
+        private void updateDisplay()
+        {
+            Scheduler.AddOnce(perform);
+
+            void perform()
+            {
+                void removeOldInfo()
```

Comment:
double nested local methods? put at the end of the parent method and don't nest this deep.

This needs reconsideration (yes I know this is in the original code, but it cannot be blindly copied into a newer version).